x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/type/DataType.java

```diff
@@ -43,7 +43,7 @@
     OBJECT(        "object",         JDBCType.STRUCT,    -1,                0,                 0,  false, false, false),
     NESTED(        "nested",         JDBCType.STRUCT,    -1,                0,                 0,  false, false, false),
     BINARY(        "binary",         JDBCType.VARBINARY, -1,                Integer.MAX_VALUE, 0,  false, false, false),
-    DATE(                            JDBCType.DATE,      Long.BYTES,        10,                10, false, false, true),
+    DATE(                            JDBCType.DATE,      Long.BYTES,        24,                24, false, false, true),
```

Comment:
Why increase the precision of date? it doesn't have a time component...

Reply:
Because we do return the whole date (e.g. 2019-03-12 00:00:00+10:00)
as DATE is internally a ZonedDateTime and we do want to show the timezone.

This was a mistake here as in the docs is correctly set to 24:
https://github.com/elastic/elasticsearch/blob/master/docs/reference/sql/language/data-types.asciidoc